### ISSUE TITLE
Display block placeholder treatments consistently when editing templates in the site editor

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#37883]
When you edit something like the single post template, blocks such as Post Title, Featured Image, and Post Content exhibit a placeholder state that helps clarify that they will entertain content from different posts.

<img width="1250" alt="Screenshot 2022-01-07 at 18 34 45" src="https://user-images.githubusercontent.com/846565/148937831-5eea5419-a9e8-4e93-b680-55995fb65ad0.png">

We should look to apply this heuristic more consistently across the template editing experience in the site editor. 

At the moment I see two key areas where this is not applied.

### Queries that inherit from template/url

<img width="1129" alt="Screenshot 2022-01-11 at 11 57 35" src="https://user-images.githubusercontent.com/846565/148938610-efbfaf97-9a0f-4afb-9a18-d598bc40baa2.png">

When you edit primary templates that include a query (e.g. Index, Archive, Search), that query displays the latest posts at the site. 

This breaks the heuristic outlined above and can mislead the user in regards to what the template will display on the frontend. Of course these templates _can_ display latest posts, but one of their defining attributes is the ability to resolve to display a variety of post lists: categories, tags, author archives, and so on.

It's extra confusing because (as you can see in the screenshot above), the Featured Image placeholder appears inline with actual content. This could lead folks to believe that the placeholder itself will render on the frontend which of course is not the case.

This should be relatively simple to address, since most blocks that appear in Queries already have placeholder states.

- [ ] When editing templates that include a query which inherits from the URL, use placeholder states for blocks inside that query

---

Side note: It might be ok for a template to render content when that template has very specific resolution conditions. For example it would probably be appropriate to see posts published by Admin in the query when editing `author-admin.html`.

### Comments loop(s)

<img width="1532" alt="Screenshot 2022-01-11 at 12 33 47" src="https://user-images.githubusercontent.com/846565/148943295-2062d305-e9c5-4ad9-8aa3-490907c41a02.png">

I'm not 100% certain of the logic, but it appears to me that any instance of the new Comments Query Loop block will simply display all comments, ordered reverse chronologically. This makes for a confusing experience when editing things like the Single Post template – you see placeholders for the post content, then actual comments below.

The blocks inside these queries should appear as placeholders. This one will be a bit more involved because I don't think the new comment blocks have placeholder states yet.

- [x] Comment Query Loop inner blocks should display placeholders when editing non-specific templates (https://github.com/WordPress/gutenberg/issues/37143)

**Comment by cbravobernal** [comment]
Great job! I completely agree with improving the template editing experience.

I will provide some context for Comments Query Loop.

By default, Comments Query Loop gets `postId` block context in order to show the comments and global `$post->id` for the frontend.

We have a research PR in #38072, where we are trying to get those default placeholders on the site editor, but we are still having small issues with the user experience on the block placeholders.

The new comment blocks have placeholders (avatar needs to improve), but they only appear if we don't have any `commentId`.

@SantosGuillamot I'm not really sure about what we could do in order to link with our tracking issue of comments, so we don't have duplicate features working on #34994

@jameskoster do you think we could apply some design to give the Comments Query Loop a nice default look? 😅 

I did some tests changing the default avatar placeholder to look like the post featured image placeholder, which right now is a text, I will post a PR as soon as possible.

**Comment by jameskoster** [comment]
> @jameskoster do you think we could apply some design to give the Comments Query Loop a nice default look? 😅

I think text-based placeholders – to be consistent with things like Post Title and so on – is fine for now :) 

For Comment Author Avatar, perhaps we could use whatever is configured as the 'Default Avatar' in the discussion settings?

**Comment by cbravobernal** [comment]
> For Comment Author Avatar, perhaps we could use whatever is configured as the 'Default Avatar' in the discussion settings

That part is done and already merged 😄

**Comment by youknowriad** [comment]
I don't really understand this issue, when you render a Query block or a comment query block, you actually have a postId, so why render a placeholder. For me it has nothing to do with other template where there's no postId.

**Comment by jameskoster** [comment]
What is the postId for a template like `tag.html`? My view is that (with a few exceptions) displaying posts that wouldn’t appear within certain generic templates can be misleading. For example I just created `tag.html` on my development site and the query renders a bunch of posts that aren't tagged at all. It gives an impression that the query is malfunctioning, and risks prompting users to set up a custom query which ironically breaks the template. It also reinforces the ongoing confusion between content and template boundaries within the Site Editor which has been a pain point since day one.

All that said this is an old issue with little activity, so obviously not high priority. Feel free to close it if you prefer.

Comments seem to have already been updated—I see placeholder comments when editing templates like Single Post.

**Comment by youknowriad** [comment]
> What is the postId for a template like tag.html? 

There's no "global post Id" in that case but the query block is a loop and for each post returned there's an actual post ID that is used.

> For example I just created tag.html on my development site and the query renders a bunch of posts that aren't tagged at all. It gives an impression that the query is malfunctioning

Ok, makes sense but for me this is a different issue. It's not really about placeholders, it's about using the right query for each template.

**Comment by jameskoster** [comment]
Yes, that is probably a better way to describe it. https://github.com/WordPress/gutenberg/issues/42554 is related. I suppose the placeholder piece is a fallback for when there is no "right" query.
